pkg/collector/corechecks/cluster/kubernetes_apiserver.go

```diff
@@ -108,6 +109,32 @@ func (k *KubeASCheck) Run() error {
 	if err != nil {
 		k.warn("could not collect API Server component status: ", err.Error())
 	}
+
+	newEvents, modifiedEvents, resVersion, err := asclient.LatestEvents(k.eventResVersion)
```

Comment:
you should check `err != nil` and throw an integration warning before using the return values, then skip the whole section.

If the watcher fails, we return `since`, so your integration warning would never show up, see https://github.com/DataDog/datadog-agent/pull/1009/files#diff-8f1c08effc368ad3aa91fdd0007818dbR47

Reply:
just to clarify on this - You mean throw a warn with the err. As off now, because the err is not logged we don't log anything ? Not sure how to use `since`

Reply:
indeed, throw an integration warning, and skip the rest (update latestEvent, aggregate, send)